flang/runtime/command.cpp

```diff
@@ -282,4 +282,40 @@ std::int32_t RTNAME(GetEnvVariable)(const Descriptor &name,
   return StatOk;
 }
 
+const char *ensureNullTerminated(
+    const char *str, size_t length, Terminator &terminator) {
+  if (length <= strlen(str)) {
```

Comment:
Please call `std::strlen` and `runtime::AllocateMemoryOrCrash` here instead.

Reply:
Done, thanks for leting me know this function, I've updated malloc used in `execute_command_line` accordingly.